extensions/skyportal/skyportal/handlers/api/alert.py

```diff
@@ -141,10 +207,313 @@ def get(self, objectId: str = None):
+          400:
+            content:
+              application/json:
+                schema: Error
+        """
+        streams = self.get_user_streams()
+
+        # allow access to public data only by default
+        selector = {1}
+
+        for stream in streams:
+            if "ztf" in stream.name.lower():
+                selector.update(set(stream.altdata.get("selector", [])))
+
+        selector = list(selector)
+
+        data = self.get_json()
+        candid = data.get("candid", None)
+        group_ids = data.pop("group_ids", None)
+
+        try:
+            query = {
+                "query_type": "aggregate",
+                "query": {
+                    "catalog": "ZTF_alerts_aux",
+                    "pipeline": [
+                        {
+                            "$match": {
+                                "_id": objectId
+                            }
+                        },
+                        {
+                            "$project": {
+                                "_id": 1,
+                                "cross_matches": 1,
+                                "prv_candidates": {
+                                    "$filter": {
+                                        "input": "$prv_candidates",
+                                        "as": "item",
+                                        "cond": {
+                                            "$in": [
+                                                "$$item.programid",
+                                                selector
+                                            ]
+                                        }
+                                    }
+                                },
+                            }
+                        },
+                        {
+                            "$project": {
+                                "_id": 1,
+                                "prv_candidates.magpsf": 1,
+                                "prv_candidates.sigmapsf": 1,
+                                "prv_candidates.diffmaglim": 1,
+                                "prv_candidates.programid": 1,
+                                "prv_candidates.fid": 1,
+                                "prv_candidates.rb": 1,
+                                "prv_candidates.ra": 1,
+                                "prv_candidates.dec": 1,
+                                "prv_candidates.candid": 1,
+                                "prv_candidates.jd": 1,
+                            }
+                        }
+                    ]
+                }
+            }
+
+            resp = await self.query_kowalski(query=query)
+
+            if resp.code == 200:
+                alert_data = tornado.escape.json_decode(resp.body).get('data', list(dict()))
+                if len(alert_data) > 0:
+                    alert_data = alert_data[0]
+                else:
+                    return self.error(f"{objectId} not found on Kowalski")
```

Comment:
As above, what about the case where someone saves an object to a group without permissions to see the data?